### ISSUE TITLE
Unable to use FusedLinear

### ISSUE THREAD

**Issue by perone** [facebookresearch/xformers#736]
I'm trying to use the `FusedLinear` from triton module but it is throwing the following error:

```
error: 'scf.for' op  along control flow edge from Region #0 to Region #0: source type #0 'tensor<64x32xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 1, warpsPe
rCTA = [2, 1]}>>' should match input type #0 'tensor<64x32xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [2, 1]}>>'
```

and 

```
Traceback (most recent call last):
  File "<string>", line 21, in kernel_fma
KeyError: ('2-.-0-.-0-317f7359f696cdaab39ac4e889e89781-d6252949da17ceb5f3a278a70250af13-3b85c7bef5f0a641282f3b73af50f599-14de7de5c4da5794c8ca14e7e41a122d-3498c340fd4b6e
e7805fd54b882a04f5-e1f133f98d04093da2078dfc51c36b72-b26258bf01f839199e39d64851821f26-d7c06e3b46e708006c15224aac7a1378-f585402118c8a136948ce0a49cfe122c', (torch.float16,
 torch.float16, torch.float16, torch.float16, torch.float16, 'i32', 'i32', 'i32', 'i32', 'i32', 'i32'), (64, 8, 32, 32, True, True, True, 1, True), (True, True, True, T
rue, True, (False, False), (True, False), (True, False), (True, False), (True, False), (True, False)))
```

I'm using PyTorch 1.12 with AMP w/ fp16 and this commit of xformers (https://github.com/facebookresearch/xformers/commit/b184fcf6e219c22958735f52c8531439258ac247). Is this fused kernel from triton really amp aware ? It looks like an issue with dtypes, but the error is very cryptic.

Without amp, using regular fp32 training, it triggers the following error as well:

```
File "/usr/local/lib/python3.8/site-packages/xformers/triton/k_fused_matmul_fw.py", line 237, in fused_matmul
    kernel_fma[grid](
  File "/usr/local/lib/python3.8/site-packages/triton/runtime/autotuner.py", line 81, in run
    self.cache[key] = builtins.min(timings, key=timings.get)
TypeError: '<' not supported between instances of 'float' and 'tuple'
```

**Comment by perone** [comment]
Update: after applying this fix (https://github.com/openai/triton/pull/1208) in Triton it works, but only if `bias=False`, w/ `bias=True` it gives the first error in the previous message.

**Comment by perone** [comment]
Closing this as the issue was related to triton version, it disappeared after fixing. Now the issue is another one related to performance, the fusedlinear is much worse than linear.

**Comment by adamcasson** [comment]
> but only if bias=False, w/ bias=True it gives the first error in the previous message.

@perone What triton version fixed this? I'm on 2.0.0 and still running into this error
